app/static/src/app/store/appState/mutations.ts

```diff
@@ -25,6 +37,36 @@ export const appStateMutations: MutationTree<AppState> = {
         state.appName = payload.appName;
         state.baseUrl = payload.baseUrl;
         state.appsPath = payload.appsPath;
+        state.language.currentLanguage = payload.defaultLanguage;
+        state.language.i18n = payload.i18n;
+        registerTranslations(state.language, {
```

Comment:
can this be moved within the translation modules, and perhaps somehow setup so that it has less duplication? Adding a new section here feels like it will be error prone